components/dashboard/src/components/Modal.tsx

```diff
@@ -75,10 +76,10 @@ export default function Modal(props: {
         <div className="fixed top-0 left-0 bg-black bg-opacity-70 z-50 w-screen h-screen">
             <div className="w-screen h-screen align-middle" style={{ display: "table-cell" }}>
                 <div
-                    className={
-                        "relative bg-white dark:bg-gray-900 border border-gray-200 dark:border-gray-800 rounded-xl p-6 max-w-lg mx-auto text-left " +
-                        (props.className || "")
-                    }
+                    className={cn(
+                        "flex flex-col max-h-screen relative bg-white dark:bg-gray-900 border border-gray-200 dark:border-gray-800 rounded-xl p-6 max-w-lg mx-auto text-left ",
```

Comment:
this allows the modal container to adjust height, while being height constrained to the screen viewport